### ISSUE TITLE
CreateKey throws NullPointerException on empty byte array

### ISSUE THREAD

**Issue by loosebazooka** [bcgit/bc-java#1370]
```
  @Test
  void test() throws IOException {
    PublicKeyFactory.createKey(new byte[0]);
  }
```
The root cause of this appears to be in [ `ASN1Primitive#fromByteArray(byte[] data)`](https://github.com/bcgit/bc-java/blob/7cc6ef8bb842682f6a008965e8ed1dffa827466a/core/src/main/java/org/bouncycastle/asn1/ASN1Primitive.java#L37) where the `ASN1inputstream` immediately returns -1 on `read()` and so the call to `readObject()` return null.

This is a bit unexpected as the null comes from deeper in the system. The input itself is not null. I don't have a good suggestion right now on the correct behavior. Perhaps it should throw an IOException?

**Comment by dghgit** [comment]
This should be dealt with in BC 1.73. Thanks for raising the issue with us.

**Comment by dghgit** [comment]
Fixed in 1.73.
